drivers/stepper/adi_tmc/adi_tmc22xx_stepper_controller.c

```diff
@@ -0,0 +1,196 @@
+
+	LOG_DBG("Stepper motor controller %s %s", dev->name, enable ? "enabled" : "disabled");
+	if (enable) {
+		return gpio_pin_set_dt(&config->enable_pin, 1);
+	} else {
+		return gpio_pin_set_dt(&config->enable_pin, 0);
+	}
+}
+
+static int tmc22xx_stepper_set_micro_step_res(const struct device *dev,
+					      enum stepper_micro_step_resolution micro_step_res)
+{
+	struct tmc22xx_data *data = dev->data;
+	const struct tmc22xx_config *config = dev->config;
+	int ret;
+
+	if (!config->msx_pins) {
+		LOG_ERR("Microstep resolution pins are not configured");
+		return -ENODEV;
+	}
+
+	for (uint8_t i = 0; i < MSX_PIN_STATE_COUNT; i++) {
+		if (micro_step_res != config->msx_resolutions[i]) {
+			continue;
+		}
+
+		ret = gpio_pin_set_dt(&config->msx_pins[0], i & 0x01);
+		if (ret < 0) {
+			LOG_ERR("Failed to set MS1 pin: %d", ret);
+			return ret;
+		}
+
+		ret = gpio_pin_set_dt(&config->msx_pins[1], (i & 0x02) >> 1);
+		if (ret < 0) {
+			LOG_ERR("Failed to set MS2 pin: %d", ret);
+			return ret;
+		}
+
+		data->resolution = micro_step_res;
+		return 0;
+	}
+
+	LOG_ERR("Unsupported microstep resolution: %d", micro_step_res);
+	return -EINVAL;
+}
+
+static int tmc22xx_stepper_get_micro_step_res(const struct device *dev,
+					      enum stepper_micro_step_resolution *micro_step_res)
+{
+	struct tmc22xx_data *data = dev->data;
+
+	*micro_step_res = data->resolution;
+	return 0;
+}
+
+static int tmc22xx_stepper_configure_msx_pins(const struct device *dev)
+{
+	const struct tmc22xx_config *config = dev->config;
+	int ret;
+
+	for (uint8_t i = 0; i < MSX_PIN_COUNT; i++) {
+		if (!gpio_is_ready_dt(&config->msx_pins[i])) {
+			LOG_ERR("MSX pin %u are not ready", i);
+			return -ENODEV;
+		}
+
+		ret = gpio_pin_configure_dt(&config->msx_pins[i], GPIO_OUTPUT);
+		if (ret < 0) {
+			LOG_ERR("Failed to configure msx pin %u", i);
+			return ret;
+		}
+	}
+	return 0;
+}
+
+static int tmc22xx_stepper_init(const struct device *dev)
+{
+	const struct tmc22xx_config *config = dev->config;
+	struct tmc22xx_data *data = dev->data;
+	int ret;
+
+	if (!gpio_is_ready_dt(&config->enable_pin)) {
+		LOG_ERR("GPIO pins are not ready");
+		return -ENODEV;
+	}
+
+	ret = gpio_pin_configure_dt(&config->enable_pin, GPIO_OUTPUT);
+	if (ret < 0) {
+		LOG_ERR("Failed to configure enable pin: %d", ret);
+		return ret;
+	}
+
+	if (config->msx_pins) {
+		ret = tmc22xx_stepper_configure_msx_pins(dev);
+		if (ret < 0) {
+			LOG_ERR("Failed to configure MSX pins: %d", ret);
+			return ret;
+		}
+
+		ret = tmc22xx_stepper_set_micro_step_res(dev, data->resolution);
+		if (ret < 0) {
+			LOG_ERR("Failed to set microstep resolution: %d", ret);
+			return ret;
+		}
+	}
+
+	ret = step_dir_stepper_common_init(dev);
+	if (ret < 0) {
+		LOG_ERR("Failed to init step dir common stepper: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static DEVICE_API(stepper, tmc22xx_stepper_api) = {
+	.enable = tmc22xx_stepper_enable,
+	.move = step_dir_stepper_common_move,
```

Comment:
How do we go about it? I think a one final force push will be required, once https://github.com/zephyrproject-rtos/zephyr/pull/82573 is merged. Sorry that this went concurrently. @fabiobaltieri any chance we can get the #82573 merged ahead of time? There is still some 11h left?